src/lib.rs

```diff
@@ -311,15 +311,64 @@ pub fn derive_from_str_as_json(input: TokenStream) -> TokenStream {
     let generics = &input.generics;
     let params = deserialize_in_generics(generics);
 
-    let result = quote! {
-        impl<#params> std::str::FromStr for #name #generics {
-            type Err = serde_json::Error;
+    let quote_string = match input.data {
+        syn::Data::Enum(enum_) => is_enum_with_simple_variants(&enum_),
+        syn::Data::Struct(syn::DataStruct {
+            fields: syn::Fields::Unnamed(fields),
+            ..
+        }) => {
+            if fields.unnamed.len() == 1 {
+                type_is_string(&fields.unnamed[0])
+            } else {
+                false
+            }
+        }
+        _ => false,
+    };
+
+    let result = if quote_string {
+        quote! {
+            impl<#params> std::str::FromStr for #name #generics {
+                type Err = serde_json::Error;
+
+                fn from_str(s: &str) -> Result<Self, Self::Err> {
+                    let s = format!(r#""{s}""#);
```

Comment:
to be completely correct, this should do json encoding of `s`, in case there are `"` and other json incompatible chars in there, but this was sufficient for my use cases.
There many already be a helper to do this in serde_json - I havent looked yet